### PR TITLE
Daemon endpoints and more

### PR DESCRIPTION
**Pull request by jr1221**
## Changes

_Explanation of changes goes here_

- Added endpoints to trigger video and serial uploads
- Added Argos/Message_Rate to the data socket

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `package-lock.json` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: scylla-server/src/controllers/file_insertion_controller.rs [hunk 1]
```diff
@@ -128,3 +134,42 @@ pub async fn insert_file(
...
 }
+
+pub async fn request_logger_insert(
+    Extension(mqtt_client): Extension<Arc<AsyncClient>>,
+) -> Result<Json<String>, ScyllaError> {
+    let mut payload = serverdata::ServerData::new();
+    payload.values = vec![1.0];
+    mqtt_client
+        .publish(
+            "Scylla/Logger/Send",
```
Comment (bracyw) [inline scylla-server/src/controllers/file_insertion_controller.rs:145]: So this is going to the car to trigger the protobuf upload right?

File: scylla-server/src/controllers/file_insertion_controller.rs [hunk 2]
```diff
@@ -128,3 +134,42 @@ pub async fn insert_file(
...
+            "Scylla/Serial/Send",
+            rumqttc::v5::mqttbytes::QoS::ExactlyOnce,
+            false,
+            protobuf::Message::write_to_bytes(&payload)
+                .unwrap_or_else(|e| format!("failed to serialize {}", e).as_bytes().to_vec()),
+        )
+        .await
+        .map_err(|err| ScyllaError::MqttError(format!("Failed to send mqtt message: {}", err)))?;
+
+    Ok(Json::from("Sent Request to insert serial logs".to_string()))
```
Comment (bracyw) [inline scylla-server/src/controllers/file_insertion_controller.rs:174]: what are the serial logs?

File: scylla-server/src/socket_handler.rs [hunk 3]
```diff
@@ -113,6 +118,25 @@ pub async fn socket_handler_with_metadata(
                         &io,
                         DATA_SOCKET_KEY,
                     ).await;
+            },
+            _ = message_rate_interval.tick() => {
+                let rate = msg_cnt as f32 / (tokio::time::Instant::now() - last_instant).as_secs() as f32;
```
Comment (bracyw) [inline scylla-server/src/socket_handler.rs]: Maybe make f64 to avoid overflow (probably not gonna happen but just in case).

**Review by bracyw — APPROVED** [review]
lgtm. Just a couple questions, the f64 thing is really up too you probably would never be a problem. WAIT ACTUALLY DO YOU WANT TO PUT THE META-DATA stuff behind a different socket key, because it seems like it will grow and it's not really part of "data" imo. Up too you, and we can do this later. Just thought of it because I was trying to test out the new meta-data and it's kind of annoying in postman with all the other non-meta spamming.

File: scylla-server/src/socket_handler.rs [hunk 4]
```diff
@@ -113,6 +118,25 @@ pub async fn socket_handler_with_metadata(
...
+                info!("Updating message rate to be {} msg/sec", rate);
+                let item = ClientData {
+                    name: "Argos/Message_Rate".to_string(),
+                    unit: "".to_string(),
+                    run_id: crate::RUN_ID.load(Ordering::Relaxed),
+                    timestamp: chrono::offset::Utc::now(),
+                    values: vec![rate]
```
Comment (bracyw) [inline scylla-server/src/socket_handler.rs:130]: maybe just input number of seconds and messages in that time for client to do calculations... that way we can always display per second

**Review by bracyw — CHANGES_REQUESTED** [review]
LOL it's been a long time since I've actually thought this is fun. I do think this comment should be impled.

File: scylla-server/src/controllers/file_insertion_controller.rs (hunk 1 above)
Comment (jr1221) [reply scylla-server/src/controllers/file_insertion_controller.rs:145]: yup

File: scylla-server/src/controllers/file_insertion_controller.rs (hunk 2 above)
Comment (jr1221) [reply scylla-server/src/controllers/file_insertion_controller.rs:174]: printf logs from the various microcontroller.s

Comment (jr1221) [reply scylla-server/src/controllers/file_insertion_controller.rs:174]: currently doesnt work lol

File: scylla-server/src/socket_handler.rs (hunk 4 above)
Comment (jr1221) [reply scylla-server/src/socket_handler.rs:130]: scylla always sends per second.  msgs/seconds always becomes msgs per second.

File: scylla-server/src/socket_handler.rs (hunk 3 above)
Comment (jr1221) [reply scylla-server/src/socket_handler.rs]: f32 is what is sent to client so it would get truncated later.

File: scylla-server/src/socket_handler.rs (hunk 4 above)
Comment (bracyw) [reply scylla-server/src/socket_handler.rs:130]: But is message interval not 5?

Comment (bracyw) [reply scylla-server/src/socket_handler.rs:130]: Oh wait nvm, is last instant only instantiated once?

Comment (bracyw) [reply scylla-server/src/socket_handler.rs:130]: Wait no I think it's per 5 seconds rn

Comment (jr1221) [reply scylla-server/src/socket_handler.rs:130]: no the interval to send is 5.  it sents the msgs/sec every 5 seconds.

**Review by bracyw — DISMISSED** [review]
Took me a second (pun intended)

**Review by bracyw — DISMISSED** [review]

**Review by bracyw — APPROVED** [review]
